src/Fmt_ast.ml

```diff
@@ -3656,8 +3673,11 @@ and fmt_value_binding c ~rec_flag ~first ?ext ?in_ ?epi ctx binding =
             $ fmt_extension_suffix c ext
             $ fmt_if_k (Option.is_some in_) (fmt_attributes c ~key:"@" atrs)
             $ fmt_if (first && Poly.(rec_flag = Recursive)) "@ rec"
-            $ fmt " " $ fmt_pattern c xpat
-            $ fmt_fun_args c ~pro:(fmt "@ ") xargs
+            $ fmt " "
+            $ ( match c.conf.fun_decl with
+              | `Compact -> hovbox 0
+              | `Sparse -> hvbox 0 )
```

Comment:
I wonder if it would be better to have a helper function to factor out these matches. Perhaps defined near `groups` to put all these choices together